src/app/clusters/application-launcher-server/application-launcher-server.cpp

```diff
@@ -192,10 +295,72 @@ bool emberAfApplicationLauncherClusterStopAppRequestCallback(app::CommandHandler
     EndpointId endpoint = commandPath.mEndpointId;
     auto & application  = commandData.application;
 
+    ApplicationLauncherApplication appInput;
+    appInput.catalogVendorId = application.catalogVendorId;
+    appInput.applicationId   = application.applicationId;
+
     Delegate * delegate = GetDelegate(endpoint);
     VerifyOrExit(isDelegateNull(delegate, endpoint) != true, err = CHIP_ERROR_INCORRECT_STATE);
 
     {
+#if CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
+        // if the feature flag is APP_Plaform, then this is a call to endpoint 1 to stop an app
+        //  1. Find target Content App
+        //  3. If this was the current app then stop it
+        //  2. Set Content App status (basic cluster) to ACTIVE_STOPPED
+        //  4. Call stop app command on Content App
+        if (delegate->HasFeature(chip::app::Clusters::ApplicationLauncher::ApplicationLauncherFeature::kApplicationPlatform))
+        {
+            ChipLogError(Zcl, "ApplicationLauncher has content platform feature");
+            ContentApp * app = chip::AppPlatform::AppPlatform::GetInstance().GetContentAppByAppId(appInput);
+            if (app == NULL)
+            {
+                ChipLogError(Zcl, "ApplicationLauncher target app not loaded");
+                Commands::LauncherResponse::Type response;
+                response.data   = chip::CharSpan("data", strlen("data"));
+                response.status = StatusEnum::kAppNotAvailable;
+                err             = command->AddResponseData(commandPath, response);
+                SuccessOrExit(err);
+            }
+            chip::app::Clusters::ApplicationBasic::Structs::ApplicationBasicApplication::Type applicationType =
```

Comment:
Again, all this extra namespacing does not need to be here, and for all sorts of bits in this function.  Please remove it to make the code easier to read.

Reply:
fixed